main/src/ca/uwaterloo/flix/api/lsp/provider/CompleteProvider.scala

```diff
@@ -33,13 +33,41 @@ object CompleteProvider {
     * Returns a list of auto-complete suggestions.
     */
   def autoComplete(uri: String, pos: Position, line: Option[String], word: Option[String])(implicit index: Index, root: TypedAst.Root): Iterable[CompletionItem] = {
-    // Ordered by priority.
-    getVarSuggestions(uri, pos, line, word) ++
-      getDefAndSigSuggestions(uri, pos, line, word) ++
-      getInstanceSuggestions(uri, pos, line, word) ++
-      getWithSuggestions(uri, pos, line, word) ++
-      getKeywordCompletionItems(line, word) ++
-      getSnippetCompletionItems(line, word)
+    word match {
+      case None => Nil;
```

Comment:
What's the idea here? I guess it is not to offer suggestions if there is no text (and vscode asks for completions).

But on the other hand, if there is no text and the user presses ctrl+space then I guess nothing happens now?

Reply:
So I am not exactly sure.

Reply:
It's more that previously the code didn't look at `word` at all, but now it does. Given that it does now look at `word` it has to do *something* when it's not defined...

Reply:
OK; I propose to structure it like this:

```
if (isAnnotationPrefix(word) { 
   ... getAnnotationCompletionItems() ...
} else {
  old block
}
```

and then add a helper function

```
def isAnnotationPrefix(word: Option[String]): Boolean
```

and go from there. 

Makes sense?